components/content/content-editor/controls/emoji/panel.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+	<div
+		class="emoji-button-container"
+		ref="container"
+		:style="{
+			top: this.top,
+		}"
+	>
+		<transition name="fade">
+			<span
+				v-if="visible"
+				class="emoji-button"
+				:class="spanClass"
+				tabindex="1"
+				@click="onButtonClick"
+				@mousedown="onMouseDown"
+				@mouseenter="onMouseEnter"
+				v-app-tooltip="panelVisible ? '' : $gettext('Insert Emoji')"
+			/>
+		</transition>
+		<transition name="fade">
+			<div
+				v-if="visible && panelVisible"
+				class="emoji-panel"
+				ref="panel"
+				@focus="onPanelFocus"
+				@blur="onPanelBlur"
+				tabindex="1"
+			>
+				<div
+					v-for="i of emojis"
```

Comment:
@hworld wants it to be like this.

Reply:
@hworld do we want to insist on using `i` as iterators everywhere if it makes the code less readable?
This made me think the iterator must be numerical when it is not.

Reply:
I only use `i` when it's a single line arrow function. If it's in a template, or a longer function, I'll name it a proper variable since the context can be too easily forgotten.